### ISSUE TITLE
chore: set up CI pipeline (GitHub Actions)

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#3]
## Context

No CI is configured. Build, lint, and test failures are only caught locally. As PRs increase, automated checks are essential to maintain quality.

## Proposed pipeline

```yaml
on: [push, pull_request]
jobs:
  check:
    - pnpm install
    - pnpm lint
    - pnpm build   # includes tsc -b
    - pnpm test
```

## Tasks

- [ ] Create `.github/workflows/ci.yml`
- [ ] Run on push to main and all PRs
- [ ] Cache pnpm store for faster runs
- [ ] Ensure Chrome types build correctly in CI (no browser needed — rules are pure functions)
- [ ] Add format check once formatter is introduced (#issue)
- [ ] Consider adding badge to README
